RetroFE/Source/Utility/Utils.cpp

```diff
@@ -79,31 +84,82 @@ std::string Utils::filterComments(std::string line)
 }
 
 
-bool Utils::findMatchingFile(const std::string& prefix, const std::vector<std::string>& extensions, std::string& file)
-{
-    namespace fs = std::filesystem; // If you're using C++17 or later
-
-    // Convert prefix to an absolute path only once
-    std::string base = Configuration::convertToAbsolutePath(Configuration::absolutePath, prefix);
+void Utils::populateCache(const std::filesystem::path& directory) {
+        Logger::write(Logger::ZONE_DEBUG, "File Cache", "Populating cache for directory: " + directory.string());
+        
+        std::unordered_set<std::string> files;
+        for (const auto& entry : std::filesystem::directory_iterator(directory)) {
+            if (entry.is_regular_file()) {
+                fileCache[directory][directory].insert(entry.path().filename().string());

```

Comment:
fileCache[directory][directory]?
fileCache[directory]

Reply:
oh yeah i had nested maps which is overly complex

> fileCache[directory][directory]? fileCache[directory]

changed to:

std::unordered_map<std::filesystem::path, std::unordered_set<std::string>> Utils::fileCache;

so now:

files.insert(entry.path().filename().string());